components/log-viewer-webui/server/src/routes/query.js

```diff
@@ -9,37 +61,36 @@ const EXTRACT_IR_TARGET_UNCOMPRESSED_SIZE = 128 * 1024 * 1024;
  * @return {Promise<void>}
  */
 const routes = async (fastify, options) => {
-    fastify.post("/query/extract-ir", async (req, resp) => {
-        const {origFileId, logEventIdx} = req.body;
+    fastify.post("/query/extract-stream", async (req, resp) => {
+        const {extractJobType, logEventIdx, streamId} = req.body;
         const sanitizedLogEventIdx = Number(logEventIdx);
-
-        let irMetadata = await fastify.dbManager.getExtractedIrFileMetadata(
-            origFileId,
+        let streamMetadata = await fastify.dbManager.getExtractedStreamFileMetadata(
+            streamId,
             sanitizedLogEventIdx
         );
 
-        if (null === irMetadata) {
-            const extractResult = await fastify.dbManager.submitAndWaitForExtractIrJob({
-                file_split_id: null,
-                msg_ix: sanitizedLogEventIdx,
-                orig_file_id: origFileId,
-                target_uncompressed_size: EXTRACT_IR_TARGET_UNCOMPRESSED_SIZE,
-            });
+        if (null === streamMetadata) {
+            await submitAndWaitForExtractStreamJob(
+                fastify,
+                extractJobType,
+                streamId,
+                sanitizedLogEventIdx
+            );
```

Comment:
so sorry for sharing a not optimal refactoring technique previously. Instead of creating a `submitAndWaitForExtractStreamJob` helper in this file, can we move the job config construction logics into the DbManager's `submitAndWaitForExtractStreamJob` method?
```suggestion
            const extractResult = await fastify.dbManager.submitAndWaitForExtractStreamJob({
                jobType: extractJobType,
                logEventIdx: sanitizedLogEventIdx,
                streamId: streamId,
                targetUncompressedSize: settings.StreamTargetUncompressedSize,
            });

            if (null === extractResult) {
                const err = new Error("Unable to extract stream with " +
            `streamId=${streamId} at logEventIdx=${sanitizedLogEventIdx}`);

                err.statusCode = 400;
                throw err;
            }

```